tests/registry.rs

```diff
@@ -581,7 +581,8 @@ fn dev_dependency_not_used() {
 fn login_with_no_cargo_dir() {
     let home = paths::home().join("new-home");
     t!(fs::create_dir(&home));
-    assert_that(cargo_process().arg("login").arg("foo").arg("-v"),
+    assert_that(cargo_process().arg("login").arg("--host").arg(registry().to_string())
```

Comment:
`--host` argument isn't required during the `login` invocation. But if it's omitted `login` assumes using `SourceId::crates_io` in order to get host and therefore updating registry from a network. It slows down the tests and requires a network connection, so I specify the fake host for the tests. (please take a look on `src/bin/login` diff for details)